cpp/src/arrow/util/align_util.cc

```diff
@@ -26,56 +26,63 @@ namespace arrow {
 
 namespace util {
 
-Result<std::shared_ptr<Array>> EnsureAlignment(const Array& object, int64_t alignment,
+Result<std::shared_ptr<Buffer>> EnsureAlignment(const std::shared_ptr<Buffer>& object, int64_t alignment,
                                                MemoryPool* memory_pool) {
```

Comment:
```suggestion
Result<std::shared_ptr<Buffer>> EnsureAlignment(std::shared_ptr<Buffer> object, int64_t alignment,
                                               MemoryPool* memory_pool) {
```

Also, the method should use move appropriate so I can do...

```
std::shared_ptr<Buffer> foo = ...
foo = EnsureAlignment(std::move(foo), 64, default_memory_pool())
// Should only make a copy of `foo` or the `shared_ptr` if the buffer was not aligned.
```

Reply:
Made the change, thanks!